src/types/dropdown-menu.ts

```diff
@@ -14,3 +14,19 @@ export const AppearanceArray = ['menu', 'selectionMenu'] as const
 export type Appearance = AnyElementOf<typeof AppearanceArray>
 
 export type DropdownItemType = 'link' | 'button' | 'default'
+
+export type DropdownItemRenderedType = 'link' | 'router-link' | 'button' | 'default'
+
+export interface IDropdownItemRenderedComponent {
+  tag: string;
+  onClick: ((event: Event) => void) | undefined;
+  attrs: {
+    class: string;
```

Comment:
question: shouldn't `class` be optional as well? It's not required to have a class on any of the components (i.e. I know they all have classes, but they don't _need_ classes)

Reply:
I left it as required because all components have it